jetstream/probe_sets.py

```diff
@@ -139,55 +141,38 @@ def discriminate_telemetry(d, type):
             for klass in type.__args__:
                 if kind == klass.kind:
                     return klass(**d)
-            raise ValueError(f"Could not discriminate telemetry kind {kind}")
+            raise ValueError(f"Could not discriminate probe kind {kind}")
 
-        converter.register_structure_hook(FeatureTelemetryType, discriminate_telemetry)
+        converter.register_structure_hook(ProbeType, discriminate_telemetry)
         return converter.structure(d, cls)
 
 
-class ResolvesFeatures(Protocol):
-    def resolve(self, slug: str) -> Feature:
+class ResolvesProbeSets(Protocol):
+    def resolve(self, slug: str) -> ProbeSet:
         ...
 
 
-class _FeatureResolver:
-    """Consume Features from the nimbus-shared repository.
-
-    This document describes how data is represented in nimbus-shared:
-    https://github.com/mozilla/nimbus-shared/blob/29526cb13c3b12ed6870ebd042261273a6e02785/docs/pages/dev/data.md
+@attr.s(auto_attribs=True)
+class _ProbeSetsResolver:
+    """Consume probe_sets from the Experimenter probesets API."""
 
-    The Feature data adopts the described convention of using a __nimbusMeta.toml
-    file in the same path as our data to describe the type of the data, so we avoid
-    consuming files with leading underscores.
-    """
+    EXPERIMENTER_API_URL_PROBESETS = "https://experimenter.services.mozilla.com/api/v6/probesets/"
+    data: Dict[str, ProbeSet] = attr.Factory(dict)
 
-    FEATURE_DEFINITION_REPO = "https://github.com/mozilla/nimbus-shared"
-    FEATURE_PATH = "data/features"
-
-    @property
-    def data(self) -> Dict[str, Feature]:
-        if data := getattr(self, "_data", None):
-            return data
-
-        with TemporaryDirectory() as tmp_dir:
-            Repo.clone_from(self.FEATURE_DEFINITION_REPO, tmp_dir)
-
-            data = {}
+    @classmethod
+    def from_experimenter(cls, session: requests.Session = None) -> "_ProbeSetsResolver":
+        session = session or requests.Session()
+        blob = session.get(cls.EXPERIMENTER_API_URL_PROBESETS).json()
+        probe_sets = {}
 
-            for spec in tmp_dir.glob(self.FEATURE_PATH + "/**/*.toml"):
-                slug = spec.stem
-                if slug.startswith("_"):
-                    continue
-                contents = toml.load(spec)
-                contents["slug"] = slug
-                feature = Feature.from_dict(contents)
-                data[slug] = feature
+        for probe_set_blob in blob:
+            probe_set = ProbeSet.from_dict(probe_set_blob)
+            probe_sets[probe_set.slug] = probe_set
 
-        self._data = data
-        return self._data
+        return cls(probe_sets)
 
-    def resolve(self, slug: str) -> Feature:
+    def resolve(self, slug: str) -> ProbeSet:
         return self.data[slug]
 
 
-FeatureResolver = _FeatureResolver()
+ProbeSetsResolver = _ProbeSetsResolver.from_experimenter()
```

Comment:
Instead of having this globally available, another option would be to pass a `_ProbeSetsResolver` instance when resolving the `AnalysisConfig`. And then also renaming `ProbeSetsResolver` to `ProbeSets`. Not sure if that would be better?

Reply:
This does work on module instantiation, which I'd like to avoid, which is why this used a lazy pattern before. Using a global object allows caching; the consumers are only allowed to read this so I think the global state is fine.

Reply:
I see, I made some modifications to keep the lazy pattern here